### DISCUSSION TITLE
Error deleting using internal hooks and db query

### DISCUSSION THREAD

**Discussion by 0x6775737461766f** [pocketbase/pocketbase#4517] [Q&A] [answered]
Hi, so i wan't to delete all records from a collection who is older than 24 hours, i tought of using record operations, but looping through each record and deleting it i don't think it is very performant, so i decided to use db.query builder, but it is throwing some error that i can't get around:

```
try {
        $app.dao().db()
        .delete()
        .from("eth_gas")
        .where($dbx.exp("created < {:date}", {date: new Date(Date.now() - 24 * 60 * 60 * 1000).toISOString()}))
    } catch (err) {
        console.log(err);
    }
```

will return

`2024/03/07 13:18:51 TypeError: Object has no member 'from'`

**Comment by ganigeorgiev** [comment, marked as answer]
The error should be clear enough.

For more details of what `delete()` accept and return please refer to https://pocketbase.io/jsvm/interfaces/dbx.Builder.html#delete.

For example: `delete("eth_gas", yourExpression).execute()`.

To help you with autocompletion and docs on hover you can also add at the top of your .pb.js file:

```js
/// <reference path="../pb_data/types.d.ts" />
```
(https://pocketbase.io/docs/js-overview/#typescript-declarations-and-code-completion)

**Reply by ganigeorgiev** [reply]
Also note that PocketBase dates are RFC3339 compatible and not ISO8601, so the date check may not work as expected. To fix it you can replace the `T` separator with blank space - `.toISOString().replace("T", " ")`.

**Reply by 0x6775737461766f** [reply]
Thank you, i was trying to use it like Knex, but now i already fixed, i also found a way of using the newQuery alongside with bind and execute to achieve my goal
